sky/templates/gcp-ray.yml.j2

```diff
@@ -116,45 +116,54 @@ rsync_exclude: []
 initialization_commands: []
 
 # List of shell commands to run to set up nodes.
+# NOTE: these are very performance-sensitive. Each new item opens/closes an SSH
+# connection, which is expensive. Try your best to co-locate commands into fewer
+# items!
+#
+# Increment the following for catching performance bugs easier:
+#   current num items (num SSH connections): 1  (+1 if tpu_vm)
 setup_commands:
-  # Create ~/.ssh/config file in case the file does not exist in the custom image.
-  # Make sure python3 & pip3 are available on this image.
+  # Line 'which conda ..': GCP TPU VM image does not install conda by default.
+  # conda needs to be a separate login, so the next item of the list has access to it.
+  {%- if tpu_vm %}
+  - pip3 install --upgrade google-api-python-client;
+    which conda > /dev/null 2>&1 || (wget -nc https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh && bash Miniconda3-latest-Linux-x86_64.sh -b && eval "$(/home/gcpuser/miniconda3/bin/conda shell.bash hook)" && conda init && conda config --set auto_activate_base true);
```

Comment:
Does this optimize the time much? This will affect the custom image, as we may want to install conda for those images.

Reply:
Good call - put conda install back and reran smoke tests (all passed except unrelated errors documenting at top), merging now.